addons/binding/org.openhab.binding.miele/ESH-INF/thing/dishwasher.xml

```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="miele"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Dishwasher -->
+	<thing-type id="dishwasher">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="xgw3000" />
+		</supported-bridge-type-refs>
+
+		<label>Dishwasher</label>
+		<description>This is a Miele@Home compatible dishwasher</description>
+
+		<channels>
+			<channel id="state" typeId="state" />
+			<channel id="program" typeId="program" />
+			<channel id="phase" typeId="phase" />
+			<channel id="start" typeId="start" />
+			<channel id="duration" typeId="duration" />
+			<channel id="elapsed" typeId="elapsed" />
+			<channel id="finish" typeId="finish" />
+			<channel id="door" typeId="door" />
+			<channel id="switch" typeId="switch" />
```

Comment:
Afaik based on the Miele protocol, they do not have any functions or commands that are mapped explicitly on physical buttons of the devices. Maybe except for the SuperFreeze on the freezers, but that's it.

On a more general note, you should know that I have implemented the protocol "as is", but that I still need to find the time to verify what each thing should do, or, how it would fit into a normal "workflow" of a device. Miele has not been 100% consistent in naming, defining, or exposing functionalities either - imo, it is quite messy.